### PR TITLE
fix #905 - correct version file template for older python versions

### LINKED ISSUE #905: generated _version.py breaks on python <3.8
I have this in my `pyproject.toml`:
```
[tool.setuptools_scm]
write_to = "src/toolname/_version.py"
```
setuptools_scm 8.0.0 seems to incorrectly generate the `_version.py` file (it contains syntax errors):
```
# file generated by setuptools_scm
# don't change, don't track in version control
from __future__ import annotations
__version__ : str = version : str = '4.5.dev38+g44c9a18'
__version_tuple__ : 'tuple[int | str, ...]' = \
  version_tuple : 'tuple[int | str, ...]' = (4, 5, 'dev38', 'g44c9a18')
```

[Logs for a failed GitHub Actions run](https://github.com/marcelm/cutadapt/actions/runs/6247483075/job/16960159583).

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
closes #905

### CODE REVIEW

File: src/setuptools_scm/_integration/dump_version.py [hunk 1]
```diff
@@ -15,10 +15,8 @@
     ".py": """\
 # file generated by setuptools_scm
 # don't change, don't track in version control
-from __future__ import annotations
-__version__ : str = version : str = {version!r}
-__version_tuple__ : 'tuple[int | str, ...]' = \\
-  version_tuple : 'tuple[int | str, ...]' = {version_tuple!r}
+__version__ = version = {version!r} # type: str
+__version_tuple__ = version_tuple = {version_tuple!r} # type: tuple[int | str, ...]
```
Comment (J535D165) [inline src/setuptools_scm/_integration/dump_version.py]:
```suggestion
__version__ = version = {version!r}  # type: str
__version_tuple__ = version_tuple = {version_tuple!r}  # type: tuple[int | str, ...]
```

Pep8 compliance.

File: testing/test_basic_api.py [hunk 2]
```diff
@@ -179,10 +179,10 @@ def read(name: str) -> str:
...
     dump_version(tmp_path, version, "first.py", scm_version=scm_version)
     lines = read("first.py").splitlines()
-    assert lines[3:] == [
-        "__version__ : str = version : str = '1.0.dev42'",
-        "__version_tuple__ : 'tuple[int | str, ...]' = \\",
-        "  version_tuple : 'tuple[int | str, ...]' = (1, 0, 'dev42')",
+    assert lines[-2:] == [
+        "__version__ = version = '1.0.dev42' # type: str",
+        "__version_tuple__ = version_tuple = (1, 0, 'dev42')"
+        " # type: tuple[int | str, ...]",
```
Comment (J535D165) [inline testing/test_basic_api.py]:
```suggestion
    assert lines[-2:] == [
        "__version__ = version = '1.0.dev42'  # type: str",
        "__version_tuple__ = version_tuple = (1, 0, 'dev42')"
        "  # type: tuple[int | str, ...]",
```

**Comment by andrewmwilson** [comment]
Thanks for the quick fix!

**Comment by RonnyPfannschmidt** [comment]
i have a pending fix for #906 as well then i will release 8.0.1

**Comment by maxnoe** [comment]
Having two type annotations in the same line is not even valid syntax in python 3.11. 

So this fix is not only for "older python versions":

```
❯ cat test_annotations.py
__version__ : str = version : str = "foo"
❯ python --version
Python 3.11.4
❯ python test_annotations.py
  File "/home/maxnoe/Uni/CTA/ctapipe/test_annotations.py", line 1
    __version__ : str = version : str = "foo"
                                ^
SyntaxError: invalid syntax
```

Valid syntax would be to split the annotated variables into one instruction per line:

```
version: str = "foo"
__version__ : str = version
```

**Pull request merged**